opm/autodiff/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -196,9 +196,23 @@ namespace Opm
             data::Wells res = WellState::report(pu);
 
             const int nw = this->numWells();
-            // If there are now wells numPhases throws a floating point
-            // exception.
-            const int np = nw ? this->numPhases() : -1;
+            if( nw == 0 ) return res;
+            const int np = pu.num_phases;
+
+
+            using rt = data::Rates::opt;
+            std::vector< rt > phs( np );
+            if( pu.phase_used[BlackoilPhases::Aqua] ) {
+                phs.at( pu.phase_pos[BlackoilPhases::Aqua] ) = rt::wat;
+            }
+
+            if( pu.phase_used[BlackoilPhases::Liquid] ) {
+                phs.at( pu.phase_pos[BlackoilPhases::Liquid] ) = rt::oil;
+            }
+
+            if( pu.phase_used[BlackoilPhases::Vapour] ) {
+                phs.at( pu.phase_pos[BlackoilPhases::Vapour] ) = rt::gas;
+            }
```

Comment:
The part above is repeated from the other function, please refactor into own function.